File: LuckyBol/app/src/main/java/com/pplus/luckybol/apps/common/builder/data/AlertData.java
package com.pplus.luckybol.apps.common.builder.data;

import com.pplus.utils.part.utils.StringUtils;
import com.pplus.luckybol.apps.common.builder.AlertBuilder;

import java.io.Serializable;
import java.util.Arrays;

/**
 * Created by j2n on 2016. 8. 22..
 */
public class AlertData implements Serializable{

    private static final long serialVersionUID = 5076033928181479319L;

    private String LOG_TAG = this.getClass().getSimpleName();

    private int identityHashCode;

    public int getIdentityHashCode(){

        return identityHashCode;
    }

    /**
     * main title
     */
    private String title;
    /**
     * contents array
     */
    private MessageData[] contents;
    /**
     * left button text
     */
    private String leftText;
    /**
     * right button text
     */
    private String rightText;
    /**
     * auto cancel
     */
    private boolean autoCancel;

    /**
     * auto cancel
     */
    private boolean backgroundClickable;

    /**
     * 기본적으로 표시되는 라인의 수 default 2line..
     */
    private int maxLine;
    /**
     * alert style..
     * <pre>
     *     기본 스타일은 Message type
     * </pre>
     */
    private AlertBuilder.STYLE_ALERT style_alert;

    public AlertData(String title, MessageData[] contents, String leftText, String rightText, boolean autoCancel, boolean backgroundClickable, AlertBuilder.STYLE_ALERT style_alert, int maxLine){

        this.title = title;
        this.contents = contents;
        this.leftText = leftText;
        this.rightText = rightText;
        this.autoCancel = autoCancel;
        this.style_alert = style_alert;
        this.backgroundClickable = backgroundClickable;
        this.maxLine = maxLine;

        this.identityHashCode = System.identityHashCode(this);
    }

    public AlertData(){

        this.identityHashCode = System.identityHashCode(this);
    }

    public String getTitle(){

        return title;
    }

    public MessageData[] getContents(){

        return contents;
    }

    public String getLeftText(){

        return leftText;
    }

    public String getRightText(){

        return rightText;
    }

    public boolean isAutoCancel(){

        return autoCancel;
    }

    public int getMaxLine(){

        return maxLine;
    }

    public boolean isSingle(){

        if(StringUtils.isEmpty(leftText) || StringUtils.isEmpty(rightText)) {
            return true;
        }
        return false;
    }

    public void setTitle(String title){

        this.title = title;
    }

    public void setContents(MessageData[] contents){

        this.contents = contents;
    }

    public void setLeftText(String leftText){

        this.leftText = leftText;
    }

    public void setRightText(String rightText){

        this.rightText = rightText;
    }

    public void setAutoCancel(boolean autoCancel){

        this.autoCancel = autoCancel;
    }

    public boolean isBackgroundClickable(){

        return backgroundClickable;
    }

    public void setBackgroundClickable(boolean backgroundClickable){

        this.backgroundClickable = backgroundClickable;
    }

    public AlertBuilder.STYLE_ALERT getStyle_alert(){

        return style_alert;
    }

    public void setStyle_alert(AlertBuilder.STYLE_ALERT style_alert){

        this.style_alert = style_alert;
    }

    @Override
    public String toString(){

        return "AlertData{" +
                "title='" + title + '\'' +
                ", contents=" + Arrays.toString(contents) +
                ", leftText='" + leftText + '\'' +
                ", rightText='" + rightText + '\'' +
                ", autoCancel=" + autoCancel +
                ", maxLine=" + maxLine +
                ", style_alert=" + style_alert +
                '}';
    }

    public static class MessageData implements Serializable{

        private static final long serialVersionUID = 6943399599090986268L;

        private String LOG_TAG = this.getClass().getSimpleName();

        public MessageData(){

        }

        private String contents;

        private AlertBuilder.MESSAGE_TYPE message_type = AlertBuilder.MESSAGE_TYPE.TEXT;

        private int maxLine = 3;

        public MessageData(String contents){

            this.contents = contents;
        }

        public MessageData(String contents, AlertBuilder.MESSAGE_TYPE message_type){

            this.contents = contents;
            this.message_type = message_type;
        }

        public MessageData(String contents, AlertBuilder.MESSAGE_TYPE message_type, int maxLine){

            this.contents = contents;
            this.message_type = message_type;
            this.maxLine = maxLine;
        }

        public void setContents(String contents){

            this.contents = contents;
        }

        public void setMessage_type(AlertBuilder.MESSAGE_TYPE message_type){

            this.message_type = message_type;
        }

        public void setMaxLine(int maxLine){

            this.maxLine = maxLine;
        }

        public String getContents(){

            return contents;
        }

        public AlertBuilder.MESSAGE_TYPE getMessage_type(){

            return message_type;
        }

        public int getMaxLine(){

            return maxLine;
        }

        @Override
        public String toString(){

            return "MessageData{" +
                    "contents='" + contents + '\'' +
                    ", message_type=" + message_type +
                    ", maxLine=" + maxLine +
                    '}';
        }
    }

    @Override
    public boolean equals(Object o){

        if(this == o) return true;
        if(o == null || getClass() != o.getClass()) return false;

        AlertData alertData = (AlertData) o;

        if(autoCancel != alertData.autoCancel) return false;
        if(maxLine != alertData.maxLine) return false;
        if(title != null ? !title.equals(alertData.title) : alertData.title != null) return false;
        // Probably incorrect - comparing Object[] arrays with Arrays.equals
        if(!Arrays.equals(contents, alertData.contents)) return false;
        return style_alert == alertData.style_alert;

    }

    @Override
    public int hashCode(){

        int result = title != null ? title.hashCode() : 0;
        result = 31 * result + Arrays.hashCode(contents);
        result = 31 * result + (autoCancel ? 1 : 0);
        result = 31 * result + maxLine;
        result = 31 * result + (style_alert != null ? style_alert.hashCode() : 0);
        return result;
    }
}

File: Buflexz/app/src/main/java/com/root37/buflexz/apps/common/builder/data/AlertData.kt
package com.root37.buflexz.apps.common.builder.data

import android.os.Parcelable
import com.pplus.utils.part.utils.StringUtils
import com.root37.buflexz.apps.common.builder.AlertBuilder.MESSAGE_TYPE
import com.root37.buflexz.apps.common.builder.AlertBuilder.STYLE_ALERT
import kotlinx.parcelize.IgnoredOnParcel
import kotlinx.parcelize.Parcelize
import java.io.Serializable
import java.util.Arrays

/**
 * Created by j2n on 2016. 8. 22..
 */

@Parcelize
class AlertData(var title: String?,
                var contents: ArrayList<MessageData>,
                var leftText: String?,
                var rightText: String?,
                var autoCancel: Boolean?,
                var backgroundClickable: Boolean?,
                var style_alert: STYLE_ALERT?,
                var maxLine: Int,
                var identityHashCode: Int = System.identityHashCode(this)) : Parcelable {
    private val LOG_TAG = this.javaClass.simpleName

    /**
     * auto cancel
     */
    var isAutoCancel = false

    /**
     * auto cancel
     */
    var isBackgroundClickable = false


    init {

        isAutoCancel = (autoCancel != null && autoCancel!!)
        isBackgroundClickable = (backgroundClickable != null && backgroundClickable!!)
    }


    fun isSingle(): Boolean {
        return StringUtils.isEmpty(leftText) || StringUtils.isEmpty(rightText)
    }



    class MessageData : Serializable {
        private val LOG_TAG = this.javaClass.simpleName

        constructor()

        var contents: String? = null
        var message_type = MESSAGE_TYPE.TEXT
        var maxLine = 3

        constructor(contents: String?) {
            this.contents = contents
        }

        constructor(contents: String?, message_type: MESSAGE_TYPE) {
            this.contents = contents
            this.message_type = message_type
        }

        constructor(contents: String?, message_type: MESSAGE_TYPE, maxLine: Int) {
            this.contents = contents
            this.message_type = message_type
            this.maxLine = maxLine
        }

        override fun toString(): String {
            return "MessageData{" + "contents='" + contents + '\'' + ", message_type=" + message_type + ", maxLine=" + maxLine + '}'
        }

        companion object {
            private const val serialVersionUID = 6943399599090986268L
        }
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val alertData = o as AlertData
        if (isAutoCancel != alertData.isAutoCancel) return false
        if (maxLine != alertData.maxLine) return false
        if (if (title != null) title != alertData.title else alertData.title != null) return false // Probably incorrect - comparing Object[] arrays with Arrays.equals
        return if (contents != alertData.contents) false else style_alert == alertData.style_alert
    }

    override fun hashCode(): Int {
        var result = if (title != null) title.hashCode() else 0
        result = 31 * result + contents.hashCode()
        result = 31 * result + if (isAutoCancel) 1 else 0
        result = 31 * result + maxLine
        result = 31 * result + if (style_alert != null) style_alert.hashCode() else 0
        return result
    }

    override fun toString(): String {
        return "AlertData(title=$title, contents=$contents, leftText=$leftText, rightText=$rightText, autoCancel=$autoCancel, backgroundClickable=$backgroundClickable, style_alert=$style_alert, maxLine=$maxLine, LOG_TAG='$LOG_TAG', identityHashCode=$identityHashCode, isAutoCancel=$isAutoCancel, isBackgroundClickable=$isBackgroundClickable)"
    }

    companion object {
        private const val serialVersionUID = 5076033928181479319L
    }
}